src/Compilers/Core/Portable/Compilation/IOperation.cs

```diff
@@ -25,6 +24,16 @@ public interface IOperation
         /// </summary>
         SyntaxNode Syntax { get; }
 
+        /// <summary>
+        /// Result type of the operation. Null if the operation is not a value-producing expression.
```

Comment:
Is it just me or this sounds like sayings all operations are expression? Maybe change to soemthing like `Null if it is not a value-producing operation` or `Null if the operation is not an expression`